app/components/edit_resource_button.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# EditResourceButton
+# edit collection btn + update work btn
+# pass in resource instead of work_version and infer type (collection or work_version)
+
+class EditResourceButton < ApplicationComponent
+  attr_reader :resource, :policy
+
+  def initialize(resource:, policy:)
+    @resource = resource
+    @policy = policy
+  end
+
+  private
```

Comment:
I'm surprised you can put all the methods as private. I would have thought that calling path, method, and label in the component template would raise "can't call private method" errors. 🤷 Anyway, I was going to suggest that those three methods be public since you're invoking them from outside the class, i.e. in the template.

Reply:
I'm guessing `private` in this case means private to the class and any templates using the class. So if you were to subclass `EditResourceButton` you would not get access to `path`, `method`, and `label`. Just a guess. But yeah, I'll move those 3 methods out of `private`.

Reply:
I'm intrigued by this now and wonder if this is a bug or if it's by design....